src/d3a/models/area/__init__.py

```diff
@@ -79,12 +112,13 @@ def __init__(self, name: str = None, children: List["Area"] = None,
         self.active = False
         self.log = TaggedLogWrapper(log, name)
         self.current_tick = 0
-        self.name = name
+        self.__name = name
         self.throughput = throughput
         self.uuid = uuid if uuid is not None else str(uuid4())
         self.slug = slugify(name, to_lower=True)
         self.parent = None
-        self.children = children if children is not None else []
+        self.children = AreaChildrenList(self, children) if children is not None\
+            else AreaChildrenList(self)
```

Comment:
```suggestion
        child_names = [c.name for c in children]
        if len(set(child_names)) !=len(child_names):
            raise Exception("Duplicate names in children")
        self.children = children
```
Even better, you can abstract it away on a validation function in order to not have a lot of things in the constructor body. IMO it will be way cleaner than overriding the Python list class.

Reply:
This makes sense to me in the creation case , but again if we try to append a child later , how can we validate the uniqueness of name

Reply:
OK now I understand the rationale behind this, thanks for bearing with me. The validation could be ported to the activate() method of the Area class (which already includes references to the parent) however this will be more messy than the solution you have implemented (because activate will also contain validation code which is irrelevant to the strategy activation). Therefore LGTM, leave it as it is.